lib/Command/Clean.php

```diff
@@ -0,0 +1,221 @@
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Tables\Command;
+
+use OCA\Tables\Db\Row;
+use OCA\Tables\Db\RowMapper;
+use OCA\Tables\Errors\InternalError;
+use OCA\Tables\Errors\NotFoundError;
+use OCA\Tables\Errors\PermissionError;
+use OCA\Tables\Service\ColumnService;
+use OCA\Tables\Service\RowService;
+use OCA\Tables\Service\TableService;
+use OCP\AppFramework\Db\DoesNotExistException;
+use OCP\AppFramework\Db\MultipleObjectsReturnedException;
+use OCP\DB\Exception;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class Clean extends Command {
+	public const PRINT_LEVEL_SUCCESS = 1;
+	public const PRINT_LEVEL_INFO = 2;
+	public const PRINT_LEVEL_WARNING = 3;
+	public const PRINT_LEVEL_ERROR = 4;
+
+	protected ColumnService $columnService;
+	protected RowService $rowService;
+	protected TableService $tableService;
+	protected LoggerInterface $logger;
+	protected RowMapper $rowMapper;
+
+	private bool $dry = false;
+	private bool $showOnlyErrors = false;
+	private int $truncateLength = 20;
+
+	private ?Row $row = null;
+	private int $offset = -1;
+
+	public function __construct(LoggerInterface $logger, ColumnService $columnService, RowService $rowService, TableService $tableService, RowMapper $rowMapper) {
+		parent::__construct();
+		$this->logger = $logger;
+		$this->columnService = $columnService;
+		$this->rowService = $rowService;
+		$this->tableService = $tableService;
+		$this->rowMapper = $rowMapper;
+	}
+
+	protected function configure(): void {
+		$this
+			->setName('tables:clean')
+			->setDescription('Clean the tables data.')
+			->addOption(
+				'dry-run',
+				'd',
+				InputOption::VALUE_NONE,
+				'Prints all wanted changes, but do not write anything to the database.'
+			)
+			->addOption(
+				'only-errors',
+				'e',
+				InputOption::VALUE_NONE,
+				'Show only errors.'
+			)
+		;
+	}
+
+	/**
+	 * @param InputInterface $input
+	 * @param OutputInterface $output
+	 * @return int
+	 */
+	protected function execute(InputInterface $input, OutputInterface $output): int {
+		$this->dry = !!$input->getOption('dry-run');
+		$this->showOnlyErrors = !!$input->getOption('only-errors');
```

Comment:
Would be nice to rather use the verbosity levels here that are provided by symfony https://symfony.com/doc/current/console/verbosity.html